main.js

```diff
@@ -1 +1,66 @@
 // write your code here to make the tests pass
+var Library = function() {
+  var books = [];                       //private array of book objects
+
+//adds a book object to the private array (created with the Book module)
+  var addBook = function(bookObject) {
+    books.push(bookObject);
+  };
+
+//only checks out if it exists and not already checked out
+  var checkOutBook = function(book) {
+    if (books.indexOf(book) > -1 && book.getAttribute("checkedOut") === false) {
```

Comment:
This is a different approach to the conditional--I like it! Though I think using a for loop would be a bit simpler and easier to read.